homeassistant/components/fritz/switch.py

```diff
@@ -0,0 +1,486 @@
+
+        # Query port forwardings and setup a switch for each forward for the current device
+        port_forwards_count: int = service_call_action(
+            fritzbox_tools, connection_type, True, "GetPortMappingNumberOfEntries"
+        )["NewPortMappingNumberOfEntries"]
+
+        _LOGGER.debug(
+            "Specific %s response: GetPortMappingNumberOfEntries=%s",
+            SWITCH_TYPE_PORTFORWARD,
+            port_forwards_count,
+        )
+
+        for i in range(port_forwards_count):
+
+            portmap = service_call_action(
+                fritzbox_tools,
+                connection_type,
+                True,
+                "GetGenericPortMappingEntry",
+                NewPortMappingIndex=i,
+            )
+
+            if not portmap:
+                return None
+
+            _LOGGER.debug(
+                "Specific %s response: GetGenericPortMappingEntry=%s",
+                SWITCH_TYPE_PORTFORWARD,
+                portmap,
+            )
+
+            # We can only handle port forwards of the given device
+            local_ip = async_get_source_ip(fritzbox_tools.host)
+            _LOGGER.debug("IP source for %s is %s", fritzbox_tools.host, local_ip)
+            if portmap["NewInternalClient"] == local_ip:
+                hass.add_job(
+                    async_add_entities,
+                    [
+                        FritzBoxPortSwitch(
+                            fritzbox_tools,
+                            entry.title,
+                            portmap,
+                            i,
+                            connection_type,
+                        )
+                    ],
+                )
+
+    def _create_profile_switches():
+        _LOGGER.debug("Setting up %s switches", SWITCH_TYPE_DEVICEPROFILE)
+        if len(fritzbox_tools.fritz_profiles) > 0:
+            for profile in fritzbox_tools.fritz_profiles.keys():
+                hass.add_job(
+                    async_add_entities,
+                    [FritzBoxProfileSwitch(fritzbox_tools, entry.title, profile)],
+                )
+
+    def _create_wifi_switches():
+        _LOGGER.debug("Setting up %s switches", SWITCH_TYPE_WIFINETWORK)
+
+        std_table = {"ac": "5Ghz", "n": "2.4Ghz"}
+        networks = {}
+        for i in range(4):
+            if ("WLANConfiguration" + str(i)) in fritzbox_tools.connection.services:
+                network_info = service_call_action(
+                    fritzbox_tools, "WLANConfiguration" + str(i), False, "GetInfo"
+                )
+                ssid = network_info["NewSSID"]
+                if ssid in networks.values():
+                    networks[i] = ssid + " " + std_table[network_info["NewStandard"]]
+                else:
+                    networks[i] = ssid
+
+        for net in networks:
+            hass.add_job(
+                async_add_entities,
+                [FritzBoxWifiSwitch(fritzbox_tools, entry.title, net, networks[net])],
+            )
+
```

Comment:
Make this its own function, and run it in the executor.  Have it return a list of entities to add and then call `async_add_entities` in async. 

You can drop all but one call to `async_add_executor_job` and all the `add_job` this way

Reply:
Done